File: hackerrank/30-days-of-code/more-exceptions/MoreExceptions.java
import java.util.*;
import java.io.*;

class Calculator{
    public int power(int base, int exponent) throws Exception{
        if ((base >= 0) && (exponent >= 0)){
            int result = (int)Math.pow(base, exponent);
            return result;
        }
        else {
            throw new Exception("n and p should be non-negative");
        }
    }
}

class MoreExceptions{

    public static void main(String[] args) {
    
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        while (t-- > 0) {
        
            int n = in.nextInt();
            int p = in.nextInt();
            Calculator myCalculator = new Calculator();
            try {
                int ans = myCalculator.power(n, p);
                System.out.println(ans);
            }
            catch (Exception e) {
                System.out.println(e.getMessage());
            }
        }
        in.close();
    }
}

File: hackerrank/30-days-of-code/more-exceptions/more-exceptions.cpp
#include<cmath>
#include<iostream>
#include<exception>
#include<stdexcept>
using namespace std;

class Calculator
{
    public:
        string s = "n and p should be non negative";
        int power(int base, int exponent){
            if(base >= 0 && exponent >= 0){
                return pow(base, exponent);
            }
            else
            {
                throw invalid_argument("n and p should be non-negative");
            }
        }

};


int main() {
    Calculator myCalculator = Calculator();
    int T, n, p;
    cin >> T;
    while(T --> 0){
        if(scanf("%d%d",&n, &p) == 2){
            try{
                int ans =myCalculator.power(n, p);
                cout << ans << endl;
            }
            catch(exception& e){
                cout << e.what() << endl;
            }
        }
    }

    return 0;
}
